website/static/js/fangorn.js

```diff
@@ -1009,6 +1020,34 @@ function _removeEvent (event, items, col) {
     }
 }
 
+function doCheckout(item, checker, error_message) {
+    $.ajax({
+        method: 'put',
+        url: window.contextVars.apiV2Prefix + 'files' + item.data.path + '/',
+        beforeSend: $osf.setXHRAuthorization,
+        contentType: 'application/json',
+        dataType: 'json',
+        data: JSON.stringify({
+            data: {
+                id: item.data.path.replace('/', ''),
+                type: 'files',
+                attributes: {
+                    checkout: checker
+                }
+            }
+        })
+    }).done(function(resp) {
+        if (error_message) {
+            window.location.reload();
+        }
+    }).fail(function(resp) {
+        if (error_message) {
```

Comment:
Is this error expected or unexpected? If unexpected, it should be logged to Sentry. See https://cosdev.readthedocs.org/en/latest/osf/guidelines.html#client-side. If expected, we should be able to give a more useful error message explaining *why* the user can't check out the file (if you can also do this for unexpected errors, all the better).